src/utils/configuration.ts

```diff
@@ -1,206 +1,26 @@
-                                    ...(info.decorators ?? []),
-                                    { name: decorator.expression.name },
-                                ];
-                            } else if (
-                                isCallExpression(decorator.expression) &&
-                                isIdentifier(decorator.expression.callee)
-                            ) {
-                                info.decorators = [
-                                    ...(info.decorators ?? []),
-                                    {
-                                        name: decorator.expression.callee.name,
-                                        arguments: parseArguments(decorator.expression.arguments),
-                                    },
-                                ];
-                            }
-
-                            existingClass.methods.push(info);
-                        }
-                    }
-                },
-            },
-        };
-    };
-
-    const require = createRequire(import.meta.url);
-    const packagePath = path.dirname(require.resolve("@babel/plugin-syntax-decorators"));
-    // const presetTypescript = path.dirname(require.resolve("@babel/preset-typescript"));
-    await babel.transformAsync(inputCode, {
-        presets: [require.resolve("@babel/preset-typescript")],
-        plugins: [
-            [packagePath, { version: "2023-05", decoratorsBeforeExport: false }],
-            extractorFunction,
-        ],
-        filename: file,
-        configFile: false,
-    });
-
-    return classes;
-}
+import { DecoratorExtractorFactory } from "./decorators/decoratorFactory.js";
 
 async function tryToReadClassInformationFromDecorators(
-    yamlBackend: Pick<YAMLBackend, "path" | "classes">,
+    yamlBackend: Pick<YAMLBackend, "path" | "classes" | "language">,
 ) {
     const cwd = yamlBackend.path || process.cwd();
 
-    const allJsFilesPaths = (await getAllFilesFromPath(cwd)).filter((file: FileDetails) => {
-        const folderPath = path.join(cwd, file.path);
-        return (
-            (file.extension === ".js" || file.extension === ".ts") &&
-            !file.path.includes("node_modules") &&
-            !file.path.includes(".git") &&
-            !fs.lstatSync(folderPath).isDirectory()
-        );
-    });
+    const decoratorExtractor = DecoratorExtractorFactory.createExtractor(yamlBackend.language.name);
+
+    const allFilesPaths = (await getAllFilesFromPath(cwd)).filter(
+        decoratorExtractor.fileFilter(cwd),
+    );
 
     return await Promise.all(
-        allJsFilesPaths.map((file) => {
+        allFilesPaths.map((file) => {
             const filePath = path.join(cwd, file.path);
-            return getDecoratorsFromFile(filePath).catch((error) => {
+            return decoratorExtractor.getDecoratorsFromFile(filePath).catch((error) => {
                 if (error.reasonCode == "MissingOneOfPlugins") {
```

Comment:
This if is specific to ts / js decorators. Maybe move it inside the `jsTsDecorator.ts`?